src/translators/sql_translator.py

```diff
@@ -543,12 +549,13 @@ def _get_where_clause(self, entity_ids, from_date, to_date, fiware_sp=None,
         if geo_clause:
             clauses.append(geo_clause)
 
-        where_clause = "where " + "and ".join(clauses)
+        if len(clauses) > 0:
+            where_clause = "where" + " and ".join(clauses)
```

Comment:
from code logic clauses will be always > 0, but if anything changes that, this would ensure that query does not break

Reply:
yep, much better!